bdai_ros2_wrappers/bdai_ros2_wrappers/executors.py

```diff
@@ -510,17 +510,30 @@ class AutoScalingMultiThreadedExecutor(rclpy.executors.Executor):
     class Task:
         """A bundle of an executable task and its associated entity."""
 
-        def __init__(self, task: rclpy.task.Task, entity: typing.Optional[rclpy.executors.WaitableEntityType]) -> None:
+        def __init__(
+            self,
+            task: rclpy.task.Task,
+            entity: typing.Optional[rclpy.executors.WaitableEntityType],
+            node: typing.Optional[rclpy.node.Node],
+        ) -> None:
             self.task = task
             self.entity = entity
+            self.valid = lambda: True
+            if node is not None and hasattr(node, "destruction_requested"):
+                self.valid = lambda: not node.destruction_requested  # type: ignore
+            else:
+                self.valid = lambda: True
```

Comment:
Do we need this else since you set this in 521?

Reply:
Oops. Leftover. Removed 521 in 3159cdcf2848645850c50bcb3e94df13765631a0